src/operator/contrib/index_copy-inl.h

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+ /*!
+  * \file index_copy-inl.h
+  * \brief implementation of index_copy tensor operation
+  */
+
+#ifndef MXNET_OPERATOR_CONTRIB_INDEX_COPY_INL_H_
+#define MXNET_OPERATOR_CONTRIB_INDEX_COPY_INL_H_
+
+#include <mxnet/operator_util.h>
+#include <vector>
+#include <limits>
+#include "../elemwise_op_common.h"
+#include "../mshadow_op.h"
+#include "../mxnet_op.h"
+
+namespace mxnet {
+namespace op {
+
+struct copy_all {
+  template<typename DType>
+  MSHADOW_XINLINE static void Map(int i,
+                                  DType* old_tensor,
+                                  DType* out_tensor) {
+    *(out_tensor + i) = *(old_tensor + i);
+  }
+};
```

Comment:
We should have a copy_all implementation. It's called identity? @eric-haibin-lin

Reply:
There is `mxnet_op::copy(ctx.get_stream<xpu>(), outputs[0], inputs[0]);`. See `static void IdentityCompute` in tensor/elemwise_unary_op.h